oak-search-elastic/src/test/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticConnectionRule.java

```diff
@@ -75,6 +80,46 @@ protected void after() {
         }
     }
 
+    public ElasticConnectionModel getElasticConnectionModel() {
+        return elasticConnectionModel;
+    }
+
+    private void initializeElasticConnectionModel(String elasticConnectionString) {
+        try {
+            URI uri = new URI(elasticConnectionString);
+            String host = uri.getHost();
+            String scheme = uri.getScheme();
+            int port = uri.getPort();
+            String query = uri.getQuery();
+
+            String apiKey = null;
+            String apiSecret = null;
+            if (query != null) {
+                apiKey = query.split(",")[0].split("=")[1];
+                apiSecret = query.split(",")[1].split("=")[1];
+            }
+            this.elasticConnectionModel = new ElasticConnectionModel();
+            elasticConnectionModel.scheme = scheme;
+            elasticConnectionModel.elasticHost = host;
+            elasticConnectionModel.elasticPort = port;
+            elasticConnectionModel.elasticApiKey = apiKey;
+            elasticConnectionModel.elasticApiSecret = apiSecret;
+            elasticConnectionModel.indexPrefix = indexPrefix + System.currentTimeMillis();
+        } catch (URISyntaxException e) {
+            LOG.error("Provided elastic connection string  is not valid ", e);
+        }
+    }
+
+    private void initializeElasticConnectionModel(ElasticsearchContainer elastic) {
+        this.elasticConnectionModel = new ElasticConnectionModel();
+        elasticConnectionModel.scheme = ElasticConnection.DEFAULT_SCHEME;
+        elasticConnectionModel.elasticHost = elastic.getContainerIpAddress();
```

Comment:
`getContainerIpAddress` is deprecated in favour of `getHost`

Reply:
Done